ios/Classes/BetterPlayerPlugin.m

```diff
@@ -251,13 +260,16 @@ - (void) setupRemoteCommandNotification:(BetterPlayer*)player, NSString* title,
                         {
                             MPMediaItemArtwork* artworkImage = [[MPMediaItemArtwork alloc] initWithImage: tempArtworkImage];
                             [_artworkImageDict setObject:artworkImage forKey:key];
-                            [nowPlayingInfoDict setObject:artworkImage forKey:MPMediaItemPropertyArtwork];
+
+                            NSMutableDictionary * currentNowPlayingInfoDict = [[MPNowPlayingInfoCenter defaultCenter].nowPlayingInfo mutableCopy];
+                            [currentNowPlayingInfoDict setObject:artworkImage forKey:MPMediaItemPropertyArtwork];
```

Comment:
I think, we should check `currentNowPlayingInfoDict` is the same content or not.

1. Content A is loaded. So the app starts to load image A.
2. Loading image A is not finished, but the user selects other content (content B). And the app starts to load image B.
3. Image B is loaded immediately. So apply it.
4. Image A is loaded. So apply it.

After the above scenario, I think the following state.
- title is content B.
- artwork is image A. (from content A.)

Is it correct?
Could you add check logic before updating the "artwork"?

Reply:
Sorry let me add something.

I could not understand the reason to make `currentNowPlayingInfoDict` by coping `nowPlayingInfo` in current code. (Is there a reason?)
I think if we make `currentNowPlayingInfoDict`, before "async" process and use it to compare with latest `nowPlayingInfo` as Ishikura san  said.

Reply:
@HiroshiOshiro 

> I could not understand the reason to make currentNowPlayingInfoDict by coping nowPlayingInfo in current code. (Is there a reason?)

Because at the time we set the "artwork", the `position` have been changed a lot. If we update the `Control center` with the `nowPlayingInfoDict` which was set at the start of `setupRemoteCommandNotification` function, the `progress bar` will jump back to the old position.

Reply:
@noboru-i cc @HiroshiOshiro 

> Is it correct?

Yes, i can reproduce your scenario.

> Could you add check logic before updating the "artwork"?

I did it, could you please check it again 🙇?

Reply:
Thanks. It seems OK 👍 

@HiroshiOshiro 
大城さんの方で確認いただき、問題なければApprove -> mergeの流れにしていただければと思います。